docs/tutorials/add-route.md

```diff
@@ -0,0 +1,163 @@
+    ->label('scope', 'files.write')
+```
+
+#### Audit
+* audits.event - Identify the log in human-readable text.
+* audits.userId - Signals the extraction of $userId in places that it's not available natively.
+* audits.resource - Signals the extraction part of the resource.
+
+
+```php
+App::post('/v1/account/create')
+->label('audits.event', 'account.create')
+->label('audits.resource', 'user/{response.$id}')
+->label('audits.userId', '{response.$id}')
+```
+
+#### Sdk
+* sdk.auth - Array of authentication types is passed in order to impose different authentication methods in different situations.
+* sdk.namespace - Refers to the route namespace.
+* sdk.method - Refers to the sdk method that needs to called.
+* sdk.description - Description of the route,using markdown format.
+* sdk.sdk.response.code - Refers to the route http response status code expected.
+* sdk.auth.response.model - Refers the route http response expected.
+
+```php
+App::post('/v1/account/jwt')
+->label('sdk.auth', [APP_AUTH_TYPE_SESSION])
+->label('sdk.namespace', 'account')
+->label('sdk.method', 'createJWT')
+->label('sdk.description', '/docs/references/account/create-jwt.md')
+->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+->label('sdk.response.model', Response::MODEL_JWT)
+```
+
+#### Cache
+* cache - When set to true, signal the use of file cache. 
+* cache.resource - Identifies the cached resource.
+
+```php
+App::get('/v1/storage/buckets/:bucketId/files/:fileId/preview')
+->label('cache', true)
+->label('cache.resource', 'file/{request.fileId}')
+```
+
+#### Abuse
+* abuse-key - Specifies routes unique abuse key.
+* abuse-limit - Specifies the number of times the route can be requested in a time frame, per route.
+* abuse-time - Specifies the time frame relevancy of the all other abuse definitions, per route.
+
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+    ->label('abuse-key', 'ip:{ip},method:{method},url:{url},userId:{userId}')
+    ->label('abuse-limit', APP_LIMIT_WRITE_RATE_DEFAULT)
+    ->label('abuse-time', APP_LIMIT_WRITE_RATE_PERIOD_DEFAULT)
+```
+
+#### Events
+* event - A pattern that is associated with the route in behalf of realtime messaging.
+  Placeholders marked as `[]` are parsed and replaced with their real values.
+
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+    ->label('event', 'buckets.[bucketId].files.[fileId].create')
+```
+
+#### Usage
+* usage.metric - The metric the route generates.
+* usage.params - Additional parameters the metrics can have.
+```php
+App::post('/v1/storage/buckets/:bucketId/files')
+    ->label('usage.metric', 'files.{scope}.requests.create')
+    ->label('usage.params', ['bucketId:{request.bucketId}'])
+```
+
+### 5. Param
+As the name implies, `param()` is used to define a request parameter.
+
+param() aspects 7 parameters :
+* A key (name) 
+* A default value
+* An instance of a relevant validator class
```

Comment:
This one can also accept a callback that returns a validator instance. Dependency injection is supported for the callback.